packages/eslint-plugin-jsx/rules/jsx-closing-bracket-location/jsx-closing-bracket-location.ts

```diff
@@ -3,10 +3,14 @@
  * @author Yannick Croissant
  */
 
+import { createRule } from '../../utils/createRule'
 import { docsUrl } from '../../utils/docsUrl'
-import report from '../../utils/report'
+import type { Tree } from '../../utils/types'
+import type { MessageIds, RuleOptions } from './types'
 
-const has = (obj, prop) => Object.prototype.hasOwnProperty.call(obj, prop)
+type InferSchemaByKey<K extends string> = Extract<RuleOptions[0], Record<K, any> | Partial<Record<K, any>>>
+
+const has = <const K extends string>(obj: object, prop: K): obj is InferSchemaByKey<K> => Object.prototype.hasOwnProperty.call(obj, prop)
```

Comment:
It would be better if we could remove this inline util

Reply:
We need to use this function to narrow the type, otherwise need to assert the type many times.